crates/bevy_utils/src/lib.rs

```diff
@@ -407,6 +407,54 @@ pub fn error<E: Debug>(result: Result<(), E>) {
     }
 }
 
+/// Boilerplate for handle wrapper
+#[macro_export]
+macro_rules! impl_handle_wrapper {
+    ($handle_wrapper_type:ident, $asset_type:ty) => {
+        impl From<$handle_wrapper_type> for AssetId<$asset_type> {
+            fn from(handle: $handle_wrapper_type) -> Self {
+                handle.id()
+            }
+        }
+
+        impl From<&$handle_wrapper_type> for AssetId<$asset_type> {
+            fn from(handle: &$handle_wrapper_type) -> Self {
+                handle.id()
+            }
+        }
+    };
+}
+
+/// Boilerplate for generic handle wrapper
+#[macro_export]
+macro_rules! impl_generic_handle_wrapper {
+    ($handle_type:ident, $asset_type:ty, $asset_type_parameter:ident, $parameter_trait:ident) => {
+        impl<$asset_type_parameter: $parameter_trait> Default
+            for $handle_type<$asset_type_parameter>
+        {
+            fn default() -> Self {
+                Self(Handle::default())
```

Comment:
As a rule of thumb, macros should use fully qualified paths to avoid issues where certain items are not in scope.